simpleclient/src/main/java/io/prometheus/client/DoubleAdder.java

```diff
@@ -118,6 +142,38 @@ public void reset() {
         internalReset(0L);
     }
 
+    public void set(double x) {
+        for (;;) {
+            Cell[] as;
+            if ((as = cells) != null) { // have cells
+                if (busy == 0 && casBusy()) {
+                    try {
+                        if (cells == as) { // recheck under lock
+                            // allocate new cells instead of updating existing cells
```

Comment:
> Thinking about it, it'd be simpler to just set the values on the Cells - but we'd still want to keep the cas on busy around for the right semantics in sum.

I don't think we could ensure correctness in `sum` with updating existing cells in `set`, as it's not possible to take a snapshot of `cells` atomically, so there would be no way to detect the intermediate state of `cells` caused by concurrent updates of individual `cells`.